packages/network/src/workers/SyncWorker.ts

```diff
@@ -108,19 +116,26 @@ export class SyncWorker<Cm extends Components> implements DoWork<SyncWorkerConfi
     const { blockNumber$ } = createBlockNumberStream(providers);
     const latestEvent$ = streamServiceUrl
       ? createLatestEventStreamService(streamServiceUrl, worldContract.address, transformWorldEvents)
-      : createLatestEventStreamRPC(blockNumber$, fetchWorldEvents);
-    const initialLiveEvents: NetworkComponentUpdate<Components>[] = [];
+      : createLatestEventStreamRPC(
+          blockNumber$,
+          fetchWorldEvents,
+          fetchSystemCalls ? createFetchSystemCallsFromEvents(provider) : undefined
+        );
 
+    const initialLiveEvents: NetworkComponentUpdate<Components>[] = [];
     latestEvent$.subscribe((event) => {
-      // If initial sync is in progress, temporary store the events to apply later
-      if (!passLiveEventsToOutput) return initialLiveEvents.push(event);
+      if (isNetworkComponentUpdateEvent(event)) {
+        // If initial sync is in progress, temporary store the events to apply later
+        if (!passLiveEventsToOutput) return initialLiveEvents.push(event);
 
-      // Store cache to indexdb every block
-      if (event.blockNumber > cacheStore.current.blockNumber + 1)
-        saveCacheStoreToIndexDb(indexDbCache, cacheStore.current);
+        // Store cache to indexdb every block
+        if (event.blockNumber > cacheStore.current.blockNumber + 1)
+          saveCacheStoreToIndexDb(indexDbCache, cacheStore.current);
+
+        storeEvent(cacheStore.current, event);
+      }
 
-      storeEvent(cacheStore.current, event);
-      this.output$.next(event as Output<Cm>);
+      this.output$.next(event as NetworkEvent<C>);
```

Comment:
Do we want to pass system calls to the main thread before the initial sync is done?